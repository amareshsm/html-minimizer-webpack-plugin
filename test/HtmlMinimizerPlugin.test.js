import {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  readAsset,
} from './helpers';

describe('HtmlMinimizerPlugin', () => {
  it('should work', async () => {
    const compiler = getCompiler({
      entry: {
        foo: `${__dirname}/fixtures/simple.js`,
      },
    });
    const stats = await compile(compiler);

    expect(readAsset('foo.js', compiler, stats)).toMatchSnapshot('result');
    expect(getErrors(stats)).toMatchSnapshot('errors');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
  });
});
