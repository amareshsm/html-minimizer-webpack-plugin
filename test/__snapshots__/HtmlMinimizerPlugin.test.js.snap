// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HtmlMinimizerPlugin should work: errors 1`] = `Array []`;

exports[`HtmlMinimizerPlugin should work: result 1`] = `
"/******/ (() => { // webpackBootstrap
/******/ 	var __webpack_modules__ = ({

/***/ \\"./simple.js\\":
/*!*******************!*\\\\
  !*** ./simple.js ***!
  \\\\*******************/
/*! unknown exports (runtime-defined) */
/*! runtime requirements: module */
/*! CommonJS bailout: module.exports is used directly at 1:0-14 */
/***/ ((module) => {

module.exports = 'html';


/***/ })

/******/ 	});
/************************************************************************/
/******/ 	// The module cache
/******/ 	var __webpack_module_cache__ = {};
/******/ 	
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/ 		// Check if module is in cache
/******/ 		if(__webpack_module_cache__[moduleId]) {
/******/ 			return __webpack_module_cache__[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = __webpack_module_cache__[moduleId] = {
/******/ 			// no module.id needed
/******/ 			// no module.loaded needed
/******/ 			exports: {}
/******/ 		};
/******/ 	
/******/ 		// Execute the module function
/******/ 		__webpack_modules__[moduleId](module, module.exports, __webpack_require__);
/******/ 	
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/ 	
/************************************************************************/
/******/ 	// startup
/******/ 	// Load entry module
/******/ 	__webpack_require__(\\"./simple.js\\");
/******/ 	// This entry module used 'module' so it can't be inlined
/******/ })()
;"
`;

exports[`HtmlMinimizerPlugin should work: warnings 1`] = `Array []`;
